data/service/api/v1/datasets_data_create.go

```diff
@@ -129,13 +129,13 @@ func DataSetsDataCreate(dataServiceContext dataService.Context) {
 }
 
 // The following 2 functions are pulled out of the above so that they can be unit tested independently.
-func MaybeUpdateSummary(ctx context.Context, registry *summary.SummarizerRegistry, updatesSummary map[string]struct{}, userId string) map[string]*time.Time {
+func MaybeUpdateSummary(ctx context.Context, registry *summary.SummarizerRegistry, updatesSummary map[string]struct{}, userId, reason string) map[string]*time.Time {
```

Comment:
This function should not be defined in `datasets_data_create.go` - it's better if it's located in the summary package (as it's also used when updating a dataset).

Reply:
done.